duckduckgo/macos-browser
# Bug Fix: Phishing Detection Dataset Discrepancies

Task/Issue URL: https://app.asana.com/0/1204023833050360/1208567121137949/f
Tech Design URL:
CC:

**Description**:
In [Implement desktop integration efficacy tests - 5-7 days](https://app.asana.com/0/1207943168535188/1207205745934704/f) it was discovered that Swift's client-side caching results in out-of-date datasets and significant dataset discrepancies between different clients. For example, it's very common for the same request to return different results from the backend, resulting in a client believing they are updating to a newer revision than they are. Over time, this compounds and results in disparate versions of the same dataset across different clients, putting users at risk of landing on newer phishing pages.

Fix: 
 - Remove Client Side Caching in PhishingDetectionClient.swift
 - Ensure embedded dataset is used to replace the on-disk dataset when the revision of the embedded dataset > on disk dataset


**Steps to test this PR**:
1. Check unit tests
3. Change on-disk revision:
4. `echo "1650000" > "/System/Volumes/Data/Users/<user>/Library/Application Support/com.duckduckgo.macos.browser.debug/revision.txt"`
5. Build the browser
6. Visit https://privacy-test-pages.site/security/badware/phishing.html
7. Ensure blocked
8. Check on-disk revision:
9. `cat "/System/Volumes/Data/Users/<user>/Library/Application Support/com.duckduckgo.macos.browser.debug/revision.txt"`
10. Should be > 1650000



**Definition of Done**:


---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
